### ISSUE TITLE
export some methods to modify the thread's name and other attribute

### ISSUE THREAD

**Issue by yanzixiang** [abcminiuser/cpptimerpool#2]
each pool create an thread under the hood,
and as you say,
Callbacks are executed on the pool's thread that created the timer
but there is no mothed to modify the thread's name and other attribute 
can you export some method to that ?

thank a lot.

**Comment by abcminiuser** [comment]
Unfortunately a std::thread does not have a standard cross platform way to adjust the thread name - there's no standard for it and each platform has it's own specific API. Adding a getter for the thread's native handle would be easy though via `std::thread::native_handle()`, allowing you to call `pthread_setname_np()` or similar on it to name it.

**Comment by yanzixiang** [comment]
```
private:
    mutable std::mutex              m_mutex;

    const std::string               m_name;

    std::forward_list<TimerHandle>  m_timers;

    std::atomic<bool>               m_running;

    std::condition_variable         m_cond;
    std::thread                     m_thread;
```

```
TimerPool::PoolHandle TimerPool::Create(const std::string& name)
{
    return std::make_shared<EnableConstructor<TimerPool>>(name);
}

TimerPool::TimerPool(const std::string& name)
    : m_mutex{ }
    , m_name{ name }
    , m_timers{ }
    , m_running{ true }
    , m_cond{ }
    , m_thread{ [this]() { run(); } }
{

}
```

so where can i get the std::thread::native_handle() as m_thread ?

https://github.com/abcminiuser/cpptimerpool/pull/3

**Comment by abcminiuser** [comment]
Sorry for the delay - I was on a work trip this week.

I did some more quick research, and of course there's another problem -- Apple's cruddy pthreads API only allows you to name the executing thread, not another thread via its handle. That means just exposing the handle isn't sufficient for something cross platform.

I've implemented the non-portable code internally in https://github.com/abcminiuser/cpptimerpool/commit/ad1b2317f088d78b3952fef69052f020088eb7ba for Windows, Linux and MacOS for now. Not ideal, but sometimes it's just not possible to do something with purely cross platform code.
